grommet/grommet
# [RadioButton] Isolate the `check` background color

#### What does this PR do?

This PR adds a new style definition to the RadioButton component.
It isolates the `check` style property, making the Button change the background color only when the radio button is checked instead of being the same color.  

#### Where should the reviewer start?

By the RadioButton component and StyledRadioButton changes.

#### What testing has been done on this PR?

Created a new test where it's verified what is the background color of the checked radio button when the `check` style prop is present.

#### How should this be manually tested?

By creating a `radioButton` theme with the `background` and `check` properties:

```js
{
  radioButton: {
    background: { // default style
      color: 'red'
    }
    check: { // only when checked
      background: {
        color: 'red'
      }
    }
  }
}
```

#### Any background context you want to provide?

The `check` property implies on a style that only applies when the radio button is checked. That was what I thought and it was the reason for separating this specific `background` property.

#### What are the relevant issues?

Closes #5225 

#### Screenshots (if appropriate)

Unchecked radio button with `background.color` prop (default `light-1`).
![Screenshot from 2021-05-11 11-53-18](https://user-images.githubusercontent.com/50644753/117837852-2bfc1280-b250-11eb-866e-692eb66f97fb.png)

Checked radio button with `check.background.color` prop.
![Screenshot from 2021-05-11 11-53-23](https://user-images.githubusercontent.com/50644753/117837869-2ef70300-b250-11eb-99e3-d5a3c38fa3d5.png)

#### Do the grommet docs need to be updated?

Yes.

#### Should this PR be mentioned in the release notes?

Yes.

#### Is this change backwards compatible or is it a breaking change?

It's not a complete breaking change. However, some radio buttons that used the `check` background theme will revert to `light-1` by default when they are not checked.
